contentcuration/contentcuration/frontend/channelEdit/views/progress/__tests__/progressModal.spec.js

```diff
@@ -1,114 +1,366 @@
 import { mount } from '@vue/test-utils';
```

Comment:
I wanted to be sure that important states of both publish and sync workflows are covered. That's why this file is separated into two parts that contain very similar tests, one for sync and another for publish operation.

I've also migrated to a more robust testing strategy where we don't reach to the internals of the component being tested as much as possible, allowing easier refactoring to me and hopefully catching more problems. Another reason is that `setMethods` is deprecated in the new version of [vue-test-utils](https://vue-test-utils.vuejs.org/api/wrapper/setmethods.html#setmethods) for the very same reason (see the deprecation warning).

We should be able to reduce the test cases to half after addressing #2969 (if we have a common component for shared tasks progress logic, it will be completely fine to test everything in detail there, and not to be so thorough in `SyncResourcesModal` and `ProgressModal` that would be built around that core).